solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -117,7 +117,7 @@ public class SearchHandler extends RequestHandlerBase
   private SolrCore core;
 
   protected List<String> getDefaultComponents() {
-    ArrayList<String> names = new ArrayList<>(8);
+    ArrayList<String> names = new ArrayList<>(9);
```

Comment:
is there any way to get rid of this magic number?   This seems very error prone.....?

Reply:
@epugh thinking about it again, it does not make too much sense to use the factory method List.of just so that we need to adapt code for dealing with a read-only list. What I would suggest instead is just initialize the ArrayList without a size at all. The default size then is applied, which is 10, we do not really save anything create a  list with an initialize size < 10.